### PR TITLE
V1.1 Lobby Communication with Clients / Game Logic

### PR DESCRIPTION
**Pull request by mhorger3**
Adding new Lobby functionality and communication standards between two clients.
Adding checks for game logic.
Adding checks for disconnected clients.
Preparing Client class for server integration

### CODE REVIEW

**Comment by mhorger3** [comment]
Will add test cases to increase coverage in V1.2

File: src/main/java/tbc/server/Lobby.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		try {
+			p1_input = new DataInputStream(players.get(0).getSocket().getInputStream());
+			p2_input = new DataInputStream(players.get(1).getSocket().getInputStream());
+			p1_output = new DataOutputStream(players.get(0).getSocket().getOutputStream());
+			p2_output = new DataOutputStream(players.get(1).getSocket().getOutputStream());
+		} catch (Exception ex) {
+			ex.printStackTrace();
+		}
+		
+		while(this.lobbyStatus == true) { // while we have a game going on
```
Comment (jzlotek) [inline src/main/java/tbc/server/Lobby.java]: maybe ```this.gameIsOn()``` or something similar

File: src/main/java/tbc/server/Lobby.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+			while(!isMoveValid) {
+				// read the move in from player one
+				try { 
+					received = p1_input.readUTF();
+				} catch (IOException ex) {
+					ex.printStackTrace();
+				}
+				
+				// if the move is valid, we must output to player two
+				if(validate(received)) {
```
Comment (jzlotek) [inline src/main/java/tbc/server/Lobby.java]: doesn't this need to be ```this.isValidMove(received)```?

File: src/main/java/tbc/server/Lobby.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+			
+			try { // output the move to player two
+				response = received;
+				p2_output.writeUTF(response);
+			} catch (IOException ex) {
+				ex.printStackTrace();
+			}
+			
+			// check if player one's move caused a winner
+			if(checkWinner())this.lobbyStatus = false;
```
Comment (jzlotek) [inline src/main/java/tbc/server/Lobby.java]: ```this.checkWinner()```

File: src/main/java/tbc/server/Lobby.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+			}		
+			try { // output the move to player one
+				response = received;
+				p1_output.writeUTF(response);
+			} catch (IOException ex) {
+				ex.printStackTrace();
+			}	
+			
+			// check if player two's move caused a winner
+			if(checkWinner())this.lobbyStatus = false;
```
Comment (jzlotek) [inline src/main/java/tbc/server/Lobby.java]: https://www.youtube.com/watch?v=M5d7vygUPoQ

File: src/main/java/tbc/server/Lobby.java (hunk 3 above)
Comment (mhorger3) [reply src/main/java/tbc/server/Lobby.java]: fixed

**Pull request merged**